clients/pubsub-common/src/test/java/org/eclipse/hono/client/pubsub/CachingPubSubPublisherFactoryTest.java

```diff
@@ -0,0 +1,115 @@
+/**
+ * Copyright (c) 2022 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.eclipse.hono.client.pubsub;
+
+import static org.mockito.Mockito.mock;
+
+import static com.google.common.truth.Truth.assertThat;
+
+import java.util.Optional;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.mockito.MockedStatic;
+import org.mockito.Mockito;
+
+/**
+ * Verifies behavior of {@link CachingPubSubPublisherFactory}.
+ */
+public class CachingPubSubPublisherFactoryTest {
+
+    private static final String TOPIC_NAME = "event";
+
+    private static final String TENANT_ID = "test-tenant";
+
+    private static final String PROJECT_ID = "test-project";
+
+    private CachingPubSubPublisherFactory factory;
+
+    @BeforeEach
+    void setUp() {
+        factory = CachingPubSubPublisherFactory.createFactory();
+    }
+
+    /**
+     * Verifies that {@link CachingPubSubPublisherFactory#getOrCreatePublisher(String, String, String)} creates a
+     * publisher and adds it to the cache.
+     */
+    @Test
+    public void testThatPublisherIsAddedToCache() {
+        assertThat(factory.getPublisher(TOPIC_NAME, TENANT_ID).isEmpty()).isTrue();
+
+        final PubSubPublisherClientImpl client = mock(PubSubPublisherClientImpl.class);
```

Comment:
The overall idea with introducing the *PubSubPublisherClient* interface was to be able to easily mock it, right?
FMPOV we should add a protected *CachingPubSubPublisherFactory.createFactory(Supplier<PubSubPublisherClient>)*  factory method which can be used in test cases to create mock clients that you can then invoke assertions on, like checking if the client's *close()* method has been invoked etc.

Reply:
From our point of view this is not necessary. We would like to keep it as it is for now.

Reply:
Ok, we can deal with that later.